modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Sorting.tsx

```diff
@@ -0,0 +1,387 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import DropDown from '@clayui/drop-down';
+import ClayForm from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayModal from '@clayui/modal';
+import {fetch, navigate, openModal, openToast} from 'frontend-js-web';
+import React, {useEffect, useRef, useState} from 'react';
+
+import {API_URL, OBJECT_RELATIONSHIP, SORTING} from '../Constants';
+import {FDSViewSectionInterface} from '../FDSView';
+import {FDSViewType} from '../FDSViews';
+import OrderableTable from '../components/OrderableTable';
+import RequiredMark from '../components/RequiredMark';
+
+type FDSSortingType = {
+	id: number;
+	name: string;
+	value: string;
+};
+
+type SortType = {
+	label: string;
+	value: string;
+};
+
+interface AddFDSSortingModalContentInterface {
+	closeModal: Function;
+	fdsView: FDSViewType;
+	onSave: Function;
+}
+
+const SORTING_OPTIONS = [SORTING.ASCENDING, SORTING.DESCENDING];
+
+const AddFDSSortingModalContent = ({
+	closeModal,
+	fdsView,
+	onSave,
+}: AddFDSSortingModalContentInterface) => {
+	const [fields, setFields] = useState<Array<string> | null>(null);
+	const [field, setField] = useState<string | null>(null);
+	const [sorting, setSorting] = useState<SortType>(SORTING_OPTIONS[0]);
+
+	useEffect(() => {
+		const getFields = async () => {
```

Comment:
[In my PR](https://github.com/liferay-frontend/liferay-portal/pull/3287) I moved the fields request to a central file, this should be fine to keep here for now, but if we merge my PR we can update this bit of code.

Reply:
> I can update the fields input to use the api.ts from https://github.com/liferay-frontend/liferay-portal/pull/3287

Just saw this 🤦